master/buildbot/reporters/github.py

```diff
@@ -135,12 +142,24 @@ def send(self, build):
         else:
             issue = None
 
-        if project:
+        if "/" in project:
             repoOwner, repoName = project.split('/')
-        else:
+        # Given we have a repository define try to get the owner and name
+        elif sourcestamps[0]['repository']:
             repo = sourcestamps[0]['repository'].split('/')[-2:]
-            repoOwner = repo[0]
-            repoName = '.'.join(repo[1].split('.')[:-1])
+            # if this repo is in the form of git@ not http://
+            if sourcestamps[0]['repository'].startswith("git@"):
```

Comment:
I think this algorithm is useful not only for github, but also for bitbucket, gitlab, and all github clones.

We should factorize it in an utility function in (create new buildbot.util.repository_urls module)
```
def getUserAndProjectFromRepoUrl(repoUrl):
     [...]
```